### ISSUE TITLE
F-Droid build failed

### ISSUE THREAD

**Issue by licaon-kter** [ivpn/android-app#260]
https://gitlab.com/fdroid/fdroiddata/-/jobs/4819480530#L482

So I added that
```
  - versionName: 2.10.0
    versionCode: 122
    commit: ff254615190d0a6a54d19fa065b7fa8ca2f404e3
    subdir: fdroid
    submodules: true
    sudo:
      - apt-get update
      - apt-get install -y make gcc g++ swig
    gradle:
      - production
    rm:
      - site
      - store
      - core/src/main/cpp/mbedtls/tests
      - core/src/main/cpp/mbedtls/crypto/tests
      - core/src/main/cpp/openvpn3/test
    prebuild: sdkmanager 'cmake;3.22.1'
    scandelete:
      - liboqs-android
    ndk: r25b
```
but...
```
...
  [98/819] Building wireguard-go
  FAILED: tools/CMakeFiles/libwg-go.so /home/vagrant/build/net.ivpn.client/core/.cxx/RelWithDebInfo/3f54295u/arm64-v8a/tools/CMakeFiles/libwg-go.so 
  cd /home/vagrant/build/net.ivpn.client/core/tools/libwg-go && make ANDROID_ARCH_NAME=arm64 ANDROID_C_COMPILER=/opt/android-sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/clang ANDROID_TOOLCHAIN_ROOT=/opt/android-sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64 ANDROID_LLVM_TRIPLE=aarch64-none-linux-android21 ANDROID_SYSROOT= ANDROID_PACKAGE_NAME= GRADLE_USER_HOME=/home/vagrant/.gradle "CFLAGS=-g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -Wno-unused-command-line-argument" "LDFLAGS=-static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--fatal-warnings -Wl,--gc-sections -Wl,--no-undefined -Qunused-arguments  -fuse-ld=gold" DESTDIR=/home/vagrant/build/net.ivpn.client/core/build/intermediates/cxx/RelWithDebInfo/3f54295u/obj/arm64-v8a BUILDDIR=/home/vagrant/build/net.ivpn.client/core/build/intermediates/cxx/RelWithDebInfo/3f54295u/obj/arm64-v8a/../generated-src
  mkdir -p "/home/vagrant/.gradle/caches/golang/"
  flock "/home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz.lock" -c ' \
  [ -f "/home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz" ] && exit 0; \
  curl -o "/home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz.tmp" "https://dl.google.com/go/go1.16.linux-amd64.tar.gz" && \
  echo "013a489ebb3e24ef3d915abe5b94c3286c070dfe0818d5bca8108f1d6e8440d2  /home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz.tmp" | sha256sum -c && \
  mv "/home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz.tmp" "/home/vagrant/.gradle/caches/golang/go1.16.linux-amd64.tar.gz"'
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   10  123M   10 12.4M    0     0  39.5M      0  0:00:03 --:--:--  0:00:03 39.5M
   83  123M   83  102M    0     0  78.2M      0  0:00:01  0:00:01 --:--:-- 78.1M
  100  123M
  internal/nettrace
  golang.zx2c4.com/wireguard/replay
  golang.org/x/crypto/internal/subtle
  unicode
  internal/race
  runtime/cgo
  internal/cpu
  runtime/internal/atomic
  sync/atomic
  runtime/internal/math
  internal/bytealg
  math
  runtime
  # runtime/cgo
  /usr/bin/ld.gold: fatal error: /opt/android-sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/lib/aarch64-linux-android/21/crtbegin_dynamic.o: unsupported ELF machine number 183
  clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
  internal/reflectlite
  sync
  internal/testlog
  internal/singleflight
  math/rand
  errors
  sort
  internal/oserror
  io
  vendor/golang.org/x/net/dns/dnsmessage
  path
  strconv
  syscall
  hash
  bytes
  strings
  crypto/hmac
  crypto
  reflect
  bufio
  internal/syscall/execenv
  internal/syscall/unix
  time
  context
  io/fs
  internal/poll
  internal/fmtsort
  encoding/binary
  os
  encoding/base64
  golang.org/x/crypto/blake2s
  golang.zx2c4.com/wireguard/tai64n
  golang.org/x/crypto/poly1305
  crypto/cipher
  golang.org/x/sys/unix
  crypto/aes
  golang.org/x/crypto/chacha20
  os/signal
  runtime/debug
  fmt
  golang.org/x/crypto/chacha20poly1305
  encoding/hex
  golang.org/x/crypto/curve25519
  golang.org/x/net/bpf
  log
  math/big
  golang.zx2c4.com/wireguard/rwcancel
  crypto/rand
  make: *** [Makefile:50: /home/vagrant/build/net.ivpn.client/core/build/intermediates/cxx/RelWithDebInfo/3f54295u/obj/arm64-v8a/libwg-go.so] Error 2
  ninja: build stopped: subcommand failed.
```

Thoughts?

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/3e06fd9ad5197af837dc2ab76db522269df5b6b7

**Comment by jurajhilje** [comment]
@licaon-kter 
Thanks for the report!
I will try to reproduce the issue with my local fdroidserver, and propose a fix asap.

**Comment by gorkapernas** [comment]
Verified on version 2.10.3, the implementation required to test login, WG key rotation, WG connections.
Tested all this without experiencing any issues.

**Comment by licaon-kter** [comment]
builds fine, fyi https://gitlab.com/fdroid/fdroiddata/-/commit/40ceba1bbe9cd8e65d906c535c71aa8231f66f22

**Comment by jurajhilje** [comment]
@licaon-kter We published a new tag/version `v2.10.3`.
The underlying issue with `v2.10.0` on F-Droid was prebuilt liboqs.so libraries. In the new version, liboqs libraries are built from source code. I was able to build the latest version with local F-Droid server.

You can find information about liboqs dependencies here:
https://github.com/ivpn/android-app/blob/5c622c75389f097ea9f7dffe6b523b9fc74db077/.github/workflows/build.yml#L27

and here:
https://github.com/open-quantum-safe/liboqs#linuxmacos

cc: @gorkapernas

**Comment by licaon-kter** [comment]
Thanks https://gitlab.com/fdroid/fdroiddata/-/commit/c6a0d9f49b42847501537f3cc50ccad3e8956cf5
